### PR TITLE
Javalab: disallow empty file names or java file names with spaces

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Prevents students from creating/renaming files with empty file names, and prevents Java file names from having spaces. Error messages are displayed like so:

Screen Shot 2021-09-16 at 5 03 37 PM
Screen Shot 2021-09-16 at 5 03 48 PM
Screen Shot 2021-09-16 at 5 03 56 PM
Screen Shot 2021-09-16 at 5 04 05 PM

## Links

https://codedotorg.atlassian.net/browse/CSA-796

## Testing story

Local & unit.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/javalab/JavalabEditorTest.js [hunk 1]
```diff
@@ -239,6 +239,64 @@ describe('Java Lab Editor Test', () => {
...
+          editTabFilename: 'Class1.java',
+          openDialog: 'renameFile',
+          orderedTabKeys: ['file-0'],
+          fileMetadata: {
+            'file-0': 'Class1.java'
+          }
+        });
+        // We are trying to rename Class1.java -> ''
+        javalabEditor.onRenameFile('');
+        expect(store.getState().javalab.renameFileError).to.not.be.null;
```
Comment (bencodeorg) [inline apps/test/unit/javalab/JavalabEditorTest.js]: Might be worth checking some of these conditions prior to your rename to confirm you're getting the expected state change?

**Review by molly-moen — APPROVED** [review]

**Review by bencodeorg — APPROVED** [review]
LGTM! Side question: do we want to allow non- .java filenames?

**Comment by sanchitmalhotra126** [comment]
> LGTM! Side question: do we want to allow non- .java filenames?

Yep, students can have non-Java files in their project. We treat them all as text files in Javabuilder.

File: apps/test/unit/javalab/JavalabEditorTest.js (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/test/unit/javalab/JavalabEditorTest.js]: Good call, will add

**Pull request merged**